### ISSUE TITLE
Full-screen `card()` accessibility and usability improvements

### ISSUE THREAD

**Issue by gadenbuie** [rstudio/bslib#957]
* Button should be revealed on `:focus` as well as hover. We'd use `opacity` to hide and reveal the button so that it's always included in the tab order.
* The button should have `aria-expanded="false"` on the full screen button.
* Full screen cards should have an `id` that's reflected in an `aria-controls="{cardId}"` attribute on the button.
* The close button in the full screen overlay should also `aria-expanded="true"` and `aria-controls` with the card ID of the current expanded card.
* This opens up the possibility of having full screen cards with a user-provided `id` become Shiny inputs that report the full screen state on that `id`.

See https://github.com/posit-dev/py-shiny/issues/951 for some inspiration of the above.

**Comment by gadenbuie** [comment]
`card(full_screen=TRUE)` doesn't currently report full screen state, here's a separate issue to track that work: https://github.com/rstudio/bslib/issues/962
